src/main/scheduler/scheduler.c

```diff
@@ -198,13 +209,25 @@ void scheduler(void)
     for (cfTask_t *task = queueFirst(); task != NULL; task = queueNext()) {
         // Task has checkFunc - event driven
         if (task->checkFunc != NULL) {
+            const uint32_t currentTimeBeforeCheckFuncCall = micros();
             // Increase priority for event driven tasks
             if (task->dynamicPriority > 0) {
                 task->taskAgeCycles = 1 + ((currentTime - task->lastSignaledAt) / task->desiredPeriod);
                 task->dynamicPriority = 1 + task->staticPriority * task->taskAgeCycles;
                 waitingTasks++;
-            } else if (task->checkFunc(currentTime, currentTime - task->lastExecutedAt)) {
-                task->lastSignaledAt = currentTime;
+            } else if (task->checkFunc(currentTimeBeforeCheckFuncCall, currentTimeBeforeCheckFuncCall - task->lastExecutedAt)) {
+#if defined(SCHEDULER_DEBUG) || !defined(SKIP_TASK_STATISTICS)
+                const uint32_t checkFuncExecutionTime = micros() - currentTimeBeforeCheckFuncCall;
+#endif
+#if defined(SCHEDULER_DEBUG)
+                DEBUG_SET(DEBUG_SCHEDULER, 3, checkFuncExecutionTime);
+#endif
+#ifndef SKIP_TASK_STATISTICS
+                checkFuncAverageExecutionTime = (checkFuncAverageExecutionTime * 31 + checkFuncExecutionTime) / 32;
```

Comment:
Rounding will cause problems here - execution time of 31us is rounded to zero ... 

Usually value*32 is stored, update is then 
```
delta = newVal - avg / 32;
avg += delta;
```